pcsx2/x86/microVU_Lower.inl

```diff
@@ -858,17 +858,50 @@ mVUop(mVU_IADDI)
 		if (_Is_ == 0)
 		{
 			const xRegister32& regT = mVU.regAlloc->allocGPR(-1, _It_, mVUlow.backupVI);
-			if (_Imm5_ != 0)
-				xMOV(regT, _Imm5_);
+			if (!EmuConfig.Gamefixes.IbitHack)
+			{
+				if (_Imm5_ != 0)
+					xMOV(regT, _Imm5_);
+				else
+					xXOR(regT, regT);
+			}
 			else
-				xXOR(regT, regT);
+			{
+				xMOV(gprT1, ptr32[&curI]);
+				xSHR(gprT1, 6);
+				xAND(gprT1, 0x1F);
+				xTEST(gprT1, 0x10);
+
+				xForwardJZ32 skipExtend;
+				xOR(gprT1, 0xFFF0);
+				skipExtend.SetTarget();
+
+				xMOVSX(regT, xRegister16(gprT1));
```

Comment:
This is just `sign_extend_5bit((*curI) >> 6))`, right?

If so, this whole sequence can be reduced to:
```asm
mov regT, dword [&curI]
shl regT, 21
sar regT, 27
```

The definition for `_Imm5_` might be worth updating too, gcc generates absolute dogcrap code, and clang is better, but still sub-optimal.